### PR TITLE
Allow timeseries writes.

### PR DESCRIPTION
**Pull request by MikeNeilson**
Modified to use content-type header for check instead of the accept
header.
Uses contentType.toString() get full content type.(if was checking
application/json against application/json;version=2 when both should've
been version=2).

Modified StoreRule Enum to generate appropriate string. The constant
value would only work if used in an actual SQL query.

- [ ] I still need to fix the openapi docs. For some reason it's completely expanding sub times like Duration. Also need to document what is and isn't required for write.

### CODE REVIEW

**Comment by MikeNeilson** [comment]
This needs an integration test but I'm also working on some tweaks to the accessmanager. I'll add those in with that PR to also test that.

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/StoreRule.java
Comment (rma-rripken) [inline cwms_radar_api/src/main/java/cwms/radar/data/dao/StoreRule.java:17]: Remove the System.out

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java
Comment (rma-rripken) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java:98]: dataum should probably be datum

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java
Comment (rma-rripken) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java:94]: I don't appreciate the usefulness of the @ArraySchema annotation.  I'm not saying this sarcastically, I really don't know what it does.  I tried these changes and one of the things that changed was that the $ref field on TimeSeries.items went away.  That $ref field makes the codegen Adam is using map our List to a Record[].  Maybe there are more annotations we can add to make the codegen work once again with the ArraySchema.  I don't know.  It seems like telling swagger that its an array of Record objects could only help but the schema that gets generated doesn't contain anything linking the field to Record.  The generated schema does describe the shape of the array items so it seems like sufficiently smart codegen could still make it work.  Once these changes get merged we should have Adam redo codegen.

Screenshot 2022-09-02 124910

File: cwms_radar_api/src/main/java/cwms/radar/api/TimeSeriesController.java
Comment (rma-rripken) [inline cwms_radar_api/src/main/java/cwms/radar/api/TimeSeriesController.java:361]: We should review the swagger docs and decide on TimeSeries, Timeseries, Time series, Time-series.  I'm definitely guilty of being inconsistent.

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java
Comment (rma-rripken) [inline cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java:45]: I assume this changed how the JSON is generated?  There is a unit test TimeSeriesTest that does some serialization tests and currently checks some of the xml fields.  This change seems like it would be easy to test in that class

Comment (MikeNeilson) [reply cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java:45]: Gah, thanks. And yes, I think I updated the json but need to verify again and make sure that's still required. There was some oddness with how the names were generated in the openapi schema vs when the formatters actually rendered it.

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java
Comment (MikeNeilson) [reply cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java:94]: I'll have to dig into that as well. It was necessary to get the example to show up correctly in the swagger ui, otherwise it doesn't render for the user correctly.
However one can modify the returned schema. so I could probably work something so that it's only rendered that was for the swagger-ui.

But hopefully there's a way to fix the ref, maybe now that I have an example that gets render it's not required.

File: cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java
Comment (MikeNeilson) [reply cwms_radar_api/src/main/java/cwms/radar/data/dto/TimeSeries.java:98]: .... oops.

File: cwms_radar_api/src/main/java/cwms/radar/data/dao/StoreRule.java
Comment (MikeNeilson) [reply cwms_radar_api/src/main/java/cwms/radar/data/dao/StoreRule.java:17]: Thank you. You know the really sad part is that probably would've shown up nicely in the checkstyle report if we didn't still have a bunch of things to fix. (Thanks for slowly fixing those each PR btw.)

**Review by rma-rripken — APPROVED** [review]

**Pull request merged**
